server/queues/tasks/DetachDocumentsFromCollectionTask.ts

```diff
@@ -27,13 +27,24 @@ export default class DetachDraftsFromCollectionTask extends BaseTask<Props> {
       return;
     }
 
+    // Detach drafts and archived documents when an unarchived collection is deleted; otherwise detach drafts only.
+    const whereOpts: WhereOptions<Document> =
```

Comment:
Looking at it with fresh eyes – this all seems like a bit of an accident waiting to happen :( – these archived documents will suddenly become available to a broader audience when the collection is deleted?

Reply:
It's controlled with this policy [here](https://github.com/outline/outline/pull/8042/files#diff-aa7db55bb651c45df93af7d14ab2a0489bb846337b4aa80e1bbb7b31cb346d5dR53). This is broader than we want it to be - I think, in this case, the document has to be restricted to the owner and admin. Thoughts?

Reply:
Yea, admin and owner would work ok in this case